Makefile.toml

```diff
@@ -172,3 +173,66 @@ dependencies = [
 
     # benchmarking and coverage jobs not included
 ]
+
+[tasks.set-nightly-version-for-ci]
+description = "Set all nightly version env vars for CI"
+category = "CI"
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+env_file = get_env GITHUB_ENV
+if not ${env_file}
+    echo "GITHUB_ENV not set!"
+    exit 1
+end
+
+echo ${env_file}
+
+needs_run = exec cargo make --loglevel error check-needs-force-nightly-ci
+
+echo ${needs_run}
+
+if eq "${needs_run}" "yes"
+    echo "Setting up CI environment for forced-nightly Rust build"
+    appendfile ${env_file} "ICU4X_NIGHTLY_TOOLCHAIN=nightly\n"
+    appendfile ${env_file} "INSTALLED_NIGHTLY_VERSION=nightly\n"
+    appendfile ${env_file} "ICU4X_BUILDING_WITH_FORCED_NIGHTLY=1\n"
+    appendfile ${env_file} "RUSTDOCFLAGS=--cfg ICU4X_BUILDING_WITH_FORCED_NIGHTLY\n"
+    exec rustup override set nightly
+else
+    echo "forced-nightly environment not required"
+end
```

Comment:
question: does this file whose path is in `GITHUB_ENV` persist across the workflow?

I wonder if we can run the boilerplate step in one job in order to initialize the values across jobs. Then have all other jobs depend on the initialization job.

Reply:
I don't think so.

You should experiment with this later, I don't want to spend too much time cleaning this up. This is _far_ from the only instance of boilerplate we have.